base/v0_2_exp/translate.go

```diff
@@ -100,3 +106,84 @@ func translateLink(from Link) (to types.Link, tm translate.TranslationSet) {
 	tm.AddIdentity("target", "hard", "overwrite", "path")
 	return
 }
+
+func translateFilesystem(from Filesystem) (to types.Filesystem, tm translate.TranslationSet) {
+	tr := translate.NewTranslator("yaml", "json")
+	tm = tr.Translate(&from.Options, &to.Options).Prefix("options")
+	to.Device = from.Device
+	to.Label = from.Label
+	to.Format = from.Format
+	to.Path = from.Path
+	to.UUID = from.UUID
+	to.WipeFilesystem = from.WipeFilesystem
+	tm.AddIdentity("device", "format", "label", "path", "uuid", "wipeFilesystem")
+	return
+}
+
+func (c Config) addMountUnits(ret *types.Config) translate.TranslationSet {
+	ts := translate.NewTranslationSet("yaml", "json")
+	if len(c.Storage.Filesystems) == 0 {
+		return ts
+	}
+	unitMap := make(map[string]int, len(ret.Systemd.Units))
+	for i, u := range ret.Systemd.Units {
+		unitMap[u.Name] = i
+	}
+	for i, fs := range c.Storage.Filesystems {
+		if fs.WithMountUnit == nil || !*fs.WithMountUnit {
+			continue
+		}
+		fromPath := path.New("yaml", "storage", "filesystems", i, "with_mount_unit")
+		newUnit := mountUnitFromFS(fs)
+		if i, ok := unitMap[unit.UnitNamePathEscape(*fs.Path)+".mount"]; ok {
+			// user also specified a unit, only set contents and enabled if the existing unit
+			// is unspecified
+			u := &ret.Systemd.Units[i]
+			unitPath := path.New("json", "systemd", "units", i)
+			if u.Contents == nil {
+				(*u).Contents = newUnit.Contents
+				ts.AddTranslation(fromPath, unitPath.Append("contents"))
+			}
+			if u.Enabled == nil {
+				(*u).Enabled = newUnit.Enabled
+				ts.AddTranslation(fromPath, unitPath.Append("enabled"))
+			}
+		} else {
+			unitPath := path.New("json", "systemd", "units", len(ret.Systemd.Units))
+			ret.Systemd.Units = append(ret.Systemd.Units, newUnit)
+			ts.AddFromCommonSource(fromPath, unitPath, newUnit)
+		}
+	}
+	return ts
+}
+
+var (
+	unitTemplate = template.Must(template.New("unit").Parse(`# Generated by FCCT
```

Comment:
Minors:
 - I generally prefer to have all module level units at the top of the individual source files right after the imports
 - Maybe name it `mountUnitTemplate`?